core/core.go

```diff
@@ -120,8 +121,18 @@ func LimaVersionSupported() error {
 		return fmt.Errorf("error decoding 'limactl info' json: %w", err)
 	}
 	// remove pre-release hyphen
-	if str := strings.SplitN(values.Version, "-", 2); len(str) > 0 {
-		values.Version = str[0]
+	parts := strings.SplitN(values.Version, "-", 2)
+	if len(parts) > 0 {
+		values.Version = parts[0]
+	}
+
+	if parts[0] == "HEAD" {
+		version := parts[0]
+		if len(parts) > 1 {
+			version = parts[1]
+		}
+		logrus.Warnf("Using a dev version of lima (%s). If the used version isn't a child of %s you may experience issues.\n", version, limaVersion)
```

Comment:
```suggestion
		logrus.Warnf("to avoid compatibility issues, ensure lima development version (%s) in use  is not lower than %s, version, limaVersion)
```

Reply:
Made this change. Just fixed some formatting in your suggestion.